Automatic creation of .deb packages (#72)
precice/calculix-adapter: packaging/README.md

# Automatic packaging of the adapter

This contains the necessary files to build a Debian package containing the adapter. (Built without PaStiX)

## Note for maintainers 

- Keep up to date the list of dependencies in the control file with `dpkg-shlibs`. Care must be taken when preCICE gets updated, typically.
- Update the changelog.

> - The script `make_deb.sh` does the required actions to compress the changelog and manpage and put them at the right place. Building the package should be as simple as running it.
> - `lintian` and `pandoc` are required to run that script. The first checks the validity of the package and the second compiles the `man` page from Markdown into the appropraite format

Reviewer: ```suggestion
- The script `make_deb.sh` compresses the changelog and manpage and puts them at the right places. Building the package should be as simple as running this script.
    - Dependencies: `lintian` (to validate the package) and `pandoc` (to generate the `man` page).
```
What I did here:
- I tried to make it more concise (deliver the same content in fewer words, more accurately)
- Replaced "running it" and "run that script" with more specific formulations. This is helpful when skimming through documentation.

# Automatic packaging of the adapter

This contains the necessary files to build a Debian package containing the adapter. (Built without PaStiX)

## Note for maintainers 

- Keep up to date the list of dependencies in the control file with `dpkg-shlibs`. In theory, it will detect preCICE with the version used for the build, so we update it manually to the lowest possible version. For instance, if the adapter is built with preCICE 2.3.0, we update the dependency as >= 2.0.0
- Update the changelog.
- The script `make_deb.sh` compresses the changelog and manpage and puts them at the right places. Building the package should be as simple as running this script.
    - Dependencies: `lintian` (to validate the package) and `pandoc` (to generate the `man` page).

> - To update CalculiX version : update the Github action workflow file (in `.github/workflows`). Update the `PACKAGE_NAME` variable, the paths (e.g. replace `wget http://www.dhondt.de/ccx_2.17.src.tar.bz2` by `wget http://www.dhondt.de/ccx_2.XX.src.tar.bz2` with XX the corresponding minor version of CalculiX and folder paths.

Reviewer: :smirk: 

```suggestion
- To update CalculiX version: update the Github action workflow file (in `.github/workflows`). Update the `PACKAGE_NAME` variable, the paths (e.g. replace `wget http://www.dhondt.de/ccx_2.17.src.tar.bz2` by `wget http://www.dhondt.de/ccx_2.XX.src.tar.bz2` with XX the corresponding minor version of CalculiX and folder paths.
```

Author: People never change :)

> - To update CalculiX version : update the Github action workflow file (in `.github/workflows`). Update the `PACKAGE_NAME` variable, the paths (e.g. replace `wget http://www.dhondt.de/ccx_2.17.src.tar.bz2` by `wget http://www.dhondt.de/ccx_2.XX.src.tar.bz2` with XX the corresponding minor version of CalculiX and folder paths.

Reviewer: Very useful file!

The version is also hard-coded in the:
- `make_deb.sh`
- `DEBIAN/control`
- `changelog.Debian`
- `manpage.md`

We should comment on this here.
